grype/db/v5/matcher/java/matcher.go

```diff
@@ -55,7 +54,10 @@ func (m *Matcher) Match(store v5.VulnerabilityProvider, d *distro.Distro, p pkg.
 	if m.cfg.SearchMavenUpstream {
 		upstreamMatches, err := m.matchUpstreamMavenPackages(store, d, p)
 		if err != nil {
-			log.Debugf("failed to match against upstream data for %s: %v", p.Name, err)
+			if strings.Contains(err.Error(), "no artifact found") {
+				log.Debugf("no upstream maven artifact found for %s", p.Name)
+			}
+			log.WithFields("package", p.Name, "error", err).Warn("failed to resolve package details with maven")
```

Comment:
@wagoodman was chatting to @luhring and he mentioned that this could still mean that errors could occur and mean results get missed.  During my testing I saw 0 failures but am wondering about the semantics of how Grype handled failures like this.  Is logging the right way to go as opposed to returning the `err`?

Reply:
Anyways, not to distract from this PR, maybe it's a followup thing.  Thanks for all the feedback so far!

Reply:
Yeah, since we're just logging the error and not returning it, I don't think this completely prevents the "silent failure" case, where vuln matches are missing from scan to scan.

My two cents is that we shouldn't let that stop us from merging this. If we start returning an error instead, we'll have a different kind of problem, even though we'd have solved silent failures.

From what I can tell, this PR is forward progress on the issue, and other work will be needed to make other kinds of improvements to the Maven lookups, like https://github.com/anchore/grype/issues/2216. And then the ideal scenario would be a mechanism where Grype could do these lookups offline.

Reply:
@kzantow and I were chatting this through -- we agree that there should be some conditions that halt and force grype to return 1 (e.g. http 429 / 503). We also think there are common cases that should be notable but not change execution (keep on matching... e.g. http 404). All of these cases should probably end up as evidence in a new `errors` section in the JSON document, and in the table output be reported as a footer noting any issues found.

I'm ok with merging this PR as is since it alleviates a direct problem and iterating towards these other more impactful suggestions as follow ons.

Reply:
fyi -- new issue created https://github.com/anchore/grype/issues/2393